src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -138,17 +141,57 @@ private static GeneratorMapData toMapData(Generator generator) {
         ActivePowerControl<Generator> activePowerControl = generator.getExtension(ActivePowerControl.class);
         if (activePowerControl != null) {
             builder.activePowerControlOn(activePowerControl.isParticipate());
+            builder.droop(activePowerControl.getDroop());
+        }
+
+        GeneratorShortCircuit generatorShortCircuit = generator.getExtension(GeneratorShortCircuit.class);
+        if (generatorShortCircuit != null) {
+            builder.transientReactance(generatorShortCircuit.getDirectTransX());
+            builder.stepUpTransformerReactance(generatorShortCircuit.getStepUpTransformerX());
+        }
+
+        GeneratorStartup generatorStartup = generator.getExtension(GeneratorStartup.class);
+        if (generatorStartup != null) {
+            builder.marginalCost(generatorStartup.getMarginalCost());
         }
 
         Terminal regulatingTerminal = generator.getRegulatingTerminal();
         //If there is no regulating terminal in file, regulating terminal voltage level is equal to generator voltage level
         if (regulatingTerminal != null && !regulatingTerminal.getVoltageLevel().equals(terminal.getVoltageLevel())) {
             builder.regulatingTerminal(regulatingTerminal.getVoltageLevel().getNameOrId());
+            builder.regulatingTerminalId(regulatingTerminal.getConnectable().getId());
+            builder.regulatingTerminalType(regulatingTerminal.getConnectable().getType().name());
+            builder.regulatingTerminalVlId(regulatingTerminal.getVoltageLevel().getId());
+        }
+
+        ReactiveLimits reactiveLimits = generator.getReactiveLimits();
+        if (reactiveLimits != null) {
+            ReactiveLimitsKind limitsKind = reactiveLimits.getKind();
+            if (limitsKind == ReactiveLimitsKind.MIN_MAX) {
+                MinMaxReactiveLimits minMaxReactiveLimits = generator.getReactiveLimits(MinMaxReactiveLimitsImpl.class);
+                builder.maximumReactivePower(minMaxReactiveLimits.getMaxQ());
+                builder.minimumReactivePower(minMaxReactiveLimits.getMinQ());
+                builder.reactiveCapabilityCurve(false);
```

Comment:
Maybe a class MinMaxReactiveLimitsMapData with minimumReactivePower and maximumReactivePower fields, added in GeneratorMapData ?
So, you could remove reactiveCapabilityCurve booleen field ?

Reply:
Ok, done